pkg/cli/kubernetes/kubernetes.go

```diff
@@ -117,6 +121,30 @@ func EnsureNamespace(ctx context.Context, client k8s.Interface, namespace string
 	return nil
 }
 
+// DeleteNamespace delete the specified namespace.
+func DeleteNamespace(ctx context.Context, client k8s.Interface, namespace string) error {
+	gracePeriodSeconds := int64(10)
+	if err := client.CoreV1().Namespaces().
+		Delete(ctx, namespace, metav1.DeleteOptions{GracePeriodSeconds: &gracePeriodSeconds}); err != nil {
+		return err
+	}
+
+	err := wait.PollUntilContextCancel(ctx, 2*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
Can you leave a comment explaining why this is needed? If this is important, we don't want someone else to come through and "simplify" it later.